lib/appsignal/span.ex

```diff
@@ -181,28 +181,107 @@ defmodule Appsignal.Span do
 
   def set_sql(_span, _body), do: nil
 
+  @spec set_params(t() | nil, map()) :: t() | nil
+  @doc """
+  Sets request parameters for an `Appsignal.Span`. Previously set
+  request parameters, such as those set by the framework, are
+  overriden.
+
+  ## Example
+      Appsignal.Tracer.root_span()
+      |> Appsignal.Span.set_params(%{"id" => 123})
+
+  """
+  def set_params(%Span{reference: _reference} = span, value)
+      when is_map(value) do
+    do_set_sample_data(span, "params", Appsignal.Utils.MapFilter.filter(value))
+  end
+
+  def set_params(_span, _value), do: nil
+
+  @spec set_environment(t() | nil, map()) :: t() | nil
+  @doc """
+  Sets the request environment data for an `Appsignal.Span`.
+  This includes the HTTP method, the request path, and the
+  request headers. A previously set request environment, such
+  as that set by the framework, is overriden.
+
+  ## Example
+      Appsignal.Tracer.root_span()
+      |> Appsignal.Span.set_headers(%{"X-Request-ID" => "a1b2c3"})
```

Comment:
```suggestion
      |> Appsignal.Span.set_headers(%{"req_headers.x_request_id" => "a1b2c3"})
```
I think this matches the format of request headers in plug?

My test apps report this format: `req_headers.accept`